src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonerIepLevelReviewService.kt

```diff
@@ -135,6 +135,7 @@ class PrisonerIepLevelReviewService(
   suspend fun getReviewById(id: Long): IepDetail =
     prisonerIepLevelRepository.findById(id)?.toIepDetail(incentiveLevelService.getAllIncentiveLevelsMapByCode()) ?: throw NoDataFoundException(id)
 
+  @Transactional
```

Comment:
adding annotation here so all effects of the received event are transactional

Reply:
BUT if `createIepForReceivedPrisoner` or `mergedPrisonerDetails` were to raise domain events, then these might be received by another party prior to the transaction committing mean they get a stale read!